vars/stageStartOpenshiftBuild.groovy

```diff
@@ -7,8 +7,23 @@ def call(def context, def buildArgs = [:], def imageLabels = [:]) {
 
     timeout(context.openshiftBuildTimeout) {
       patchBuildConfig(context, buildArgs, imageLabels)
-      sh "oc start-build ${context.componentId} --from-dir docker --follow -n ${context.targetProject}"
+      sh (script: "oc start-build ${context.componentId} --from-dir docker --follow -n ${context.targetProject}", label : "start openshift build")
     }
+
+    def ocpbuild = sh(returnStdout: true, script:"oc get build --sort-by=.status.startTimestamp --no-headers -n ${context.targetProject} -l buildconfig=${context.componentId} | tail -n1", label : "find last build").trim().split(/\s+/)
+    
+    def ocpbuildId = ocpbuild[0]
+    def ocpbuildStatus = ocpbuild[3]
+    
+    if (ocpbuildStatus.toString().trim().toLowerCase() != "complete") {
+      error "OCP Build ${ocpbuildId} was not successfull - status ${ocpbuildStatus}"
+    }
+  
+    def ocpCurrentImage = sh(returnStdout: true, script:"oc get istag ${context.componentId}:${context.getTagversion()} -n ${context.targetProject} --no-headers", label : "find new image").trim().split(/\s+/)
```

Comment:
I would like to see this (and the other examples) replaced with JSONpath expressions. Eg.
```
oc get istag -n foo-dev be-main:1-1598e8f1 -o jsonpath='{.image.dockerImageReference}'
```

I believe this is:
1. easier to read
2. more stable as the path is part of the API, whereas the CLI output is (I think) subject to change

Reply:
@michaelsauter  - top will do .. (i wish the json engine would be good enough for more complex stuff) - see comment above ..